x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshots.java

```diff
@@ -185,6 +187,32 @@
         Setting.Property.PrivateIndex,
         Setting.Property.NotCopyableOnResize
     );
+    public static final String SNAPSHOT_BLOB_CACHE_METADATA_FILES_MAX_LENGTH = "index.store.snapshot.blob_cache.metadata_files.max_length";
+    public static final Setting<ByteSizeValue> SNAPSHOT_BLOB_CACHE_METADATA_FILES_MAX_LENGTH_SETTING = new Setting<>(
+        new Setting.SimpleKey(SNAPSHOT_BLOB_CACHE_METADATA_FILES_MAX_LENGTH),
+        s -> new ByteSizeValue(64L, ByteSizeUnit.KB).getStringRep(),
+        s -> Setting.parseByteSize(
+            s,
+            new ByteSizeValue(1L, ByteSizeUnit.KB),
+            new ByteSizeValue(Long.MAX_VALUE),
+            SNAPSHOT_BLOB_CACHE_METADATA_FILES_MAX_LENGTH
+        ),
+        value -> {
+            if (value.getBytes() % BufferedIndexInput.BUFFER_SIZE != 0L) {
+                final String message = String.format(
+                    Locale.ROOT,
+                    "failed to parse value [%s] for setting [%s], must be a multiple of [%s] bytes",
+                    value.getStringRep(),
+                    SNAPSHOT_BLOB_CACHE_METADATA_FILES_MAX_LENGTH,
+                    BufferedIndexInput.BUFFER_SIZE
+                );
+                throw new IllegalArgumentException(message);
+            }
+        },
+        Setting.Property.IndexScope,
+        Setting.Property.NodeScope,
```

Comment:
Let's make it purely indexScope?

Reply:
Good catch, I pushed b5f2463ba4c80ddf4d87d969e969aced2d70f849